Index: nsXULStub.cpp
===================================================================
RCS file: /cvsroot/mozilla/xulrunner/stub/nsXULStub.cpp,v
retrieving revision 1.11
diff -u -8 -p -r1.11 nsXULStub.cpp
--- nsXULStub.cpp	29 Nov 2007 17:20:54 -0000	1.11
+++ nsXULStub.cpp	15 Dec 2007 00:33:52 -0000
@@ -168,17 +168,17 @@ main(int argc, char **argv)
   CFURLRef absResourcesURL = CFURLCopyAbsoluteURL(resourcesURL);
   CFRelease(resourcesURL);
   if (!absResourcesURL)
     return 1;
 
   CFURLRef iniFileURL =
     CFURLCreateCopyAppendingPathComponent(kCFAllocatorDefault,
                                           absResourcesURL,
-                                          CFSTR("application.ini"),
+                                          CFSTR("../../../application.ini"),
                                           false);
   CFRelease(absResourcesURL);
   if (!iniFileURL)
     return 1;
 
   CFStringRef iniPathStr =
     CFURLCopyFileSystemPath(iniFileURL, kCFURLPOSIXPathStyle);
   CFRelease(iniFileURL);
@@ -243,17 +243,17 @@ main(int argc, char **argv)
   if (!lastSlash)
     return 1;
 
   *(++lastSlash) = '\0';
 
   // On Linux/Win, look for XULRunner in appdir/xulrunner
 
   snprintf(greDir, sizeof(greDir),
-           "%sxulrunner" XPCOM_FILE_PATH_SEPARATOR XPCOM_DLL,
+           "%sxulrunner-linux" XPCOM_FILE_PATH_SEPARATOR XPCOM_DLL,
            iniPath);
 
   greFound = (access(greDir, R_OK) == 0);
 
   strncpy(lastSlash, "application.ini", sizeof(iniPath) - (lastSlash - iniPath));
 
 #endif
 
